epersonregistrations.md

```diff
@@ -0,0 +1,58 @@
+# EPerson registration Endpoints
+[Back to the list of all defined endpoints](endpoints.md)
+
+## Main Endpoint
+**GET /api/eperson/epersonregistrations**
+
+As we don't have a use case to iterate over all the existent epersonregistrations, the main endpoint is not implemented and a 405 error code is returned according to our [general error response codes](README.md#Error-codes).
+
+## Single EPerson registration
+
+As we don't have a use case to retrieve an epersonregistration based on the email, the single endpoint is not implemented and a 405 error code is returned according to our [general error response codes](README.md#Error-codes).
+
+## Search EPerson registration
+**/api/eperson/epersonregistrations/search/findByToken?token=<:token>**
+
+Exposes the registered email address based on the token.  
+Also exposes whether it's a new user registration, or a password reset for an existing user.
+
+```json
+{
+  "email": "user@institution.edu",
+  "user": null,
+  "type": "epersonregistration"
+}
+```
+
+```json
+{
+  "email": "user@institution.edu",
+  "user": "028dcbb8-0da2-4122-a0ea-254be49ca107",
+  "type": "epersonregistration"
+}
+```
+
+## Create new EPerson registration
+**POST /api/eperson/epersonregistrations**
+
+To create a new EPerson registration, perform a post with the JSON below to the eperson registrations endpoint (without being authenticated).
+
+```json
+{
+  "email": "user@institution.edu",
+  "type": "epersonregistration"
+}
+```
+
+No other properties can be set (e.g. the name cannot be defined)
+If successful, an email will be sent with a token allowing the user to continue the registration
+
+Status codes:
+* 201 Created - if the operation succeed
+* 422 Unprocessable Entity - if the email address was omitted
```

Comment:
In the XMLUI, it was possible to disable registration altogether: https://github.com/DSpace/DSpace/blob/master/dspace/config/dspace.cfg#L1836

Do we want to bring forward a similar configuration?  If so, there should be an error code here that notes that registration is disabled / not allowed.  (I do know some sites liked this feature, as they chose to not allow anyone to create an account....instead accounts were only allowed through either LDAP or Shibb or similar)

Although if we choose to keep this configuration, disabling registration probably should *not* also disable "forgot password".

Reply:
I've included an error message for this, and how the UI can verify whether the feature is enabled